vimdoc/output.py

```diff
@@ -56,12 +56,22 @@ def WriteTableOfContents(self):
     """Writes the table of contents."""
     self.WriteRow()
     self.WriteLine('CONTENTS', right=self.Tag(self.Slug('contents')))
-    for i, block in enumerate(self.module.sections.values()):
+    # We need to keep track of section numbering on a per-level basis
```

Comment:
Is it possible to extract out a generator function as a helper that tracks the indexing? Then the code in this method could look like:
```python
for index, block in self._EnumerateIndices(self.module.sections.values()):
  self.WriteLine('%d. %s' % (index, block.locals['name']),
                 indent=2 * block.locals['level'] + 1,
                 right=self.Link(self.Slug(block.locals['id'])),
                 fill='.')
```